### PR TITLE
Allow disabling directory crawling for tests

### PR DESCRIPTION
**Pull request by stevearc**
Intended to address #13

Relatively straightforward: adds a config option that, when disabled, doesn't call `_update_positions` on directories. Files still get populated in the summary on `BufOpen`/`BufWritePost`, and I also added some logic to scan the files that vim opens on start.

## Test plan

New functionality
- Set `discovery.enabled = false`
- Open a test file in a directory with multiple test files
- Observe that the test summary only displays the currently-open file

Nothing broke
- Set `discovery.enabled = true`
- Open a test file in a directory with multiple test files
- Observe that the test summary displays all discoverable tests, as per usual

## Future work
Per-directory configuration would still be a great addition. Another feature that could be cool is some test adapters are capable of running tests on a directory. It would be great if we could trigger a run on an entire directory without scanning it first, and then populate the summary based on the results. I may start looking into that next.

### CODE REVIEW

**Comment by rcarriga** [comment]
Thanks for putting this together! I'll probably have time over the weekend to review properly :+1:

**Comment by stevearc** [comment]
Figured out the issue with tests not updating. Will address in a separate PR

**Comment by stevearc** [comment]
Oh I didn't mention this directly, but this PR is ready for review. It had some unfortunate side effects before the position merging, but now it works fine.

File: lua/neotest/client/init.lua
Comment (rcarriga) [inline lua/neotest/client/init.lua:578]: Let's add a test for this case.

**Review by rcarriga — CHANGES_REQUESTED** [review]
This is very nice, and works brilliantly with the new merging code. Thanks for putting this together!

Just want to add the one test case and then LGTM

**Comment by stevearc** [comment]
Done!

**Pull request merged**
